include/deal.II/matrix_free/matrix_free.templates.h

```diff
@@ -373,6 +373,26 @@ namespace internal
                             (cell->level(), cell->index()));
       }
   }
+
+
+
+  // For serial Triangulations, we simply take all cells.
+  // In this case the subdomain_id does not correspond to the process id
+  // Hence, it is not necessarily constant.
+  template <typename InIterator>
+  void resolve_cell_serial (const InIterator   &cell,
+                            std::vector<std::pair<unsigned int,unsigned int> > &cell_its)
+  {
+    if (cell->has_children())
+      for (unsigned int child=0; child<cell->n_children(); ++child)
+        resolve_cell_serial (cell->child(child), cell_its);
+    else
+      {
+        Assert (cell->active(), ExcInternalError());
+        cell_its.push_back (std::pair<unsigned int,unsigned int>
+                            (cell->level(), cell->index()));
+      }
+  }
```

Comment:
Wouldn't it be possible to route this code through `resolve_cell` as well with `subdomain_id = numbers::invalid_subdomain_id`? This would avoid the duplication here and simply add a different argument to the function in case you cannot cast to `parallel::Triangulation`.